chroma_core/lib/job.py

```diff
@@ -201,6 +201,34 @@ def invoke_agent(self, host, command, args={}):
             self._log_subprocesses(e.subprocesses)
             raise
 
+    def invoke_rust_local_action(self, command, args={}):
+        """
+        Talks to the iml-action-runner service
+        """
+
+        return invoke_rust_local_action(command, args, self._cancel_event)
+
+    def invoke_rust_local_action_expect_result(self, command, args={}):
+        from chroma_core.services.job_scheduler.agent_rpc import LocalActionException
+
+        try:
+            result = self.invoke_rust_local_action(command, args)
+        except RustAgentCancellation as e:
+            raise LocalActionException(command, args, "Cancelled: {}".format(e))
+        except Exception as e:
+            raise LocalActionException(command, args, "Unexpected error: {}".format(e))
+
+        try:
+            result = json.loads(result)
+        except ValueError as e:
```

Comment:
Is it only ever throwing `ValueError`? Shouldn't we handle other possible exceptions?

Reply:
If there is a circular reference it could throw an `OverflowError`. It could also throw a `TypeError` but I think `ValueError` is really the most common. https://docs.python.org/2.7/library/json.html

Reply:
> If there is a circular reference it could throw an OverflowError

Given this is *deserializing* a JSON string, I'm not sure how an `OverflowError` is possible.

Also if we reach this point, we are basically guaranteed to be dealing with `bytes`, so I don't think TypeError applies either.

Long story short, `ValueError` is the only exception we should worry about here.